internal/k8s/reconciler/route_test.go

```diff
@@ -266,6 +266,20 @@ func TestRouteMergedStatusAndBinding(t *testing.T) {
 	require.Equal(t, "expected", statuses[0].Conditions[0].Message)
 	require.Equal(t, RouteConditionReasonBindError, statuses[0].Conditions[0].Reason)
 
+	// check route ref
+	route = NewK8sRoute(inner, K8sRouteConfig{
+		ControllerName: "expected",
+		Logger:         hclog.NewNullLogger(),
+	})
+	route.resolutionErrors.Add(service.NewRefNotPermittedError("not found"))
+	route.OnBindFailed(nil, gateway)
+	statuses = route.MergedStatus().Parents
+	require.Len(t, statuses, 3)
+	require.Equal(t, "expected", string(statuses[0].ParentRef.Name))
```

Comment:
It's helpful to use the `assert` package whenever failing the assertion doesn't invalidate the rest of the test. In cases like this, it's helpful to know if the later assertions pass/fail even if this one fails, and `require` immediately ends the test on a failure.